modules/ai/assets/js/editor/pages/form-layout/index.js

```diff
@@ -141,6 +158,30 @@ const FormLayout = ( { onClose, onInsert, onData, onSelect, onGenerate, DialogHe
 		};
 	};
 
+	/**
+	 * @param {Attachment[]} items
+	 */
+	const onAttach = ( items ) => {
+		items.forEach( ( item ) => {
+			if ( ! attachmentsTypes[ item.type ] ) {
+				throw new Error( `Invalid attachment type: ${ item.type }` );
+			}
+
+			const typeConfig = attachmentsTypes[ item.type ];
+
+			if ( ! item.previewHTML && typeConfig.previewGenerator ) {
+				typeConfig.previewGenerator( item.content ).then( ( html ) => {
+					item.previewHTML = html;
+
+					setAttachments( () => [ item ] );
+				} );
+			}
+		} );
+
+		setAttachments( items );
+		setIsPromptEditable( true );
```

Comment:
I would probably put it above the loop (even though it's technically almost the same)
unless you actually care about it not being called if there was an error on line 167

Reply:
It should run only if there are no errors